hamocc/mo_read_shelfmask.F90

```diff
@@ -0,0 +1,133 @@
+! Copyright (C) 2021-2022  J. Schwinger
+!
+! This file is part of BLOM/iHAMOCC.
+!
+! BLOM is free software: you can redistribute it and/or modify it under the
+! terms of the GNU Lesser General Public License as published by the Free
+! Software Foundation, either version 3 of the License, or (at your option)
+! any later version.
+!
+! BLOM is distributed in the hope that it will be useful, but WITHOUT ANY
+! WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+! FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for
+! more details.
+!
+! You should have received a copy of the GNU Lesser General Public License
+! along with BLOM. If not, see https://www.gnu.org/licenses/.
+
+
+module mo_read_shelfmask
+
+  !*************************************************************************************************
+  ! Routine to read and/or initialize an ocean shelf-sea mask either from netcdf file or from
+  ! internal bathymetry values, if use_shelfsea_res_time = .true.
+  !
+  ! If a file is read, it should hold real values with 1. for shelf-sea regions and 0 elsewhere
+  ! and the variable name is 'shelfmask'
+  !
+  ! The shelf-sea mask is currently used to calculate the shelf-sea water residence time
+  !
+  ! j.maerz *UiB, Bergen* 2024-10-31
+  !
+  !*************************************************************************************************
+
+  implicit none
+  private
+
+  public :: read_shelfmask
+
+  character(len=512),               public :: shelfsea_maskfile =''
```

Comment:
I tried this now locally and then the namelist reading fails to correctly initialize the file name (it silently ignores the entry and leaves the file name as empty - and just runs further). Maybe we should introduce a string-length parameter to be able to set those things at one place... - up for later.

Reply:
OK, no problem, just leave it as it is.

Reply:
@TomasTorsvik , not sure, if this can be extended to a full namelist: https://stackoverflow.com/questions/14886787/fortran-character-input-at-undefined-length - for now, I keep the old style.